### DISCUSSION TITLE
Update and Delete of Source Data

### DISCUSSION THREAD

**Discussion by alberk8** [ClickHouse/ClickHouse#61120] [Q&A]
I understand that ClickHouse does not support native UPDATE statement.  How does it handle data coming from SQL sources (like postgres) where data row can be update or deleted?.

**Comment by den-crane** [comment]
It uses Replacing or Collapsing engines and `from ... FINAL` queries

https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/replacingmergetree
https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/collapsingmergetree
https://clickhouse.com/docs/en/sql-reference/statements/select/from#final-modifier

**Reply by alberk8** [reply]
Does that mean that update does work on MergeTree Table?

**Comment by yiyang-shao** [comment]
I recommend exploring the detailed guide [ClickHouse PostgreSQL Change Data Capture (CDC) - Part 1](https://clickhouse.com/blog/clickhouse-postgresql-change-data-capture-cdc-part-1#replacingmergetree). available on ClickHouse's official blog to see whether it fits your use case and answers your question. This resource specifically discusses the use of ReplacingMergeTree engine to efficiently manage data updates and replacements.
